src/d3a/models/strategy/area_agents/two_sided_pay_as_clear_engine.py

```diff
@@ -60,7 +60,7 @@ def _discrete_point_curve(self, obj):
             if len(obj) <= 1 or i == (len(obj) - 1):
                 break
             rate = math.floor(obj[i+1].price / obj[i+1].energy)
-            cumulative[rate] += obj[i].energy
+            cumulative[rate] += obj[i+1].energy
```

Comment:
Wait, I am confused here. Wasn't this loop refactored before? I think after a comment of @hannesdiedrich you replaced the index-based loop to iterating over the items without using indices. If I am imagining this, then this needs to be refactored, it looks much better and it is much safer if the indices are not used.

Reply:
Yes, you are right. It is in branch 932.1 (https://github.com/gridsingularity/d3a/pull/368/files#diff-9727365ead9d7d435176c7f1c9a53c19R55)